test/e2e/framework/helper/validate.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2020 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"context"
+
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ValidationFunc describes a certificate validation helper function
+type ValidationFunc func(certificate *cmapi.Certificate, secret *v1.Secret) error
+
+// ValidateCertificate retreives the issued certificate and runs all validation functions
+func (h *Helper) ValidateCertificate(validations []ValidationFunc, ns, name string) error {
```

Comment:
Would it make sense for this function to return a `wait.ConditionFunc`? Or provide some easy way to produce functions that do?

It'd then read something like:

```
wait.Poll(helper.ValidateCertificate([]helper.ValidationFunc{ ... }, "namespace", "name"), timeout, interval)
```

which feels like a fairly